src/scripts/sync_synonym.py

```diff
@@ -84,13 +88,46 @@ def _add_synonym_type_inferences(row: pd.Series, exclusions: Set[str]) -> str:
     return '|'.join(types)
 
 
-def _add_syn_variation_cols(df: pd.DataFrame) -> pd.DataFrame:
-    """Set 'synonym' col, and add cols showing capitalization differences in between Mondo and source."""
+def _handle_synonym_casing_variations(df: pd.DataFrame) -> pd.DataFrame:
+    """Add and modify fields related to synonym capitalization variations"""
+    # Set 'synonym' col
     df['synonym'] = df['synonym_case_mondo']
+    # Add cols showing capitalization differences in between Mondo and source.
     df['synonym_case_diff_mondo'] = df.apply(lambda row:
         row['synonym_case_mondo'] if row['synonym_case_mondo'] != row['synonym_case_source'] else '', axis=1)
     df['synonym_case_diff_source'] = df.apply(lambda row:
         row['synonym_case_source'] if row['synonym_case_source'] != row['synonym_case_mondo'] else '', axis=1)
+
+    # Capitalization variations for single source term synonym: drop dupe rows & aggregate
+    agg_dict: Dict[str, Union[str, Callable]] = {col: 'first' for col in df.columns}
+    agg_dict['synonym_case_source'] = '|'.join
+    df = df.groupby(['mondo_id', 'source_id', 'synonym_scope', 'synonym'], as_index=False).agg(agg_dict)
+    df['synonym_case_source_is_many'] = df['synonym_case_source'].apply(lambda x: '|' in x)
+    # - Correct synonym_case_diff_source field for these cases
+    df['synonym_case_diff_source'] = df.apply(lambda row: row['synonym_case_source']
+        if row['synonym_case_source_is_many'] else row['synonym_case_diff_source'], axis=1)
+
+    # Capitalization variations for single Mondo term synonym: keep 2+ rows, but aggregate variations in 'diff' cols
```

Comment:
## Bug: 3 instances where `synonym` column was mutated
I observed that the `|` delimited multi-synonym values were happening in this column, rather than `synonym_case_diff_mondo `. Not sure why this is happening. Need to fix.

<img width="1048" alt="Screenshot 2024-12-16 at 6 30 35 PM" src="https://github.com/user-attachments/assets/7e1641b7-adfc-4fc0-b5e4-5edaf829eb5e" />

- [ ] @joeflack4 fix

Reply:
@joeflack4 is this fixed?

Reply:
This is fixed with a new pull request I just opened

Reply:
We should merge that pr first, though I will put in a tweak and explanation momentarily

Reply:
Which PR?

Reply:
- #724

Reply:
@joeflack4 when I run `sync-synonyms` from this feature branch I get an error
```
mkdir -p reports/sync-synonym
robot --catalog catalog-v001.xml query -i components/doid.owl --query ../sparql/synonyms-scope-type-xref.sparql tmp/doid-synonyms-scope-type-xref.tsv
python3 ../scripts/sync_synonym.py \
--mondo-mappings-path tmp/mondo.sssom.tsv \
--ontology-db-path components/doid.db \
--mondo-synonyms-path tmp/mondo-synonyms-scope-type-xref.tsv \
--mondo-exclusion-configs config/mondo-exclusion-configs.yml \
--onto-synonym-types-path tmp/doid-synonyms-scope-type-xref.tsv \
--onto-config-path metadata/doid.yml \
--outpath-added reports/sync-synonym/doid.synonyms.added.robot.tsv \
--outpath-confirmed reports/sync-synonym/doid.synonyms.confirmed.robot.tsv \
--outpath-updated reports/sync-synonym/doid.synonyms.updated.robot.tsv
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/dist-packages/pandas/core/indexes/base.py", line 3805, in get_loc
    return self._engine.get_loc(casted_key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "index.pyx", line 167, in pandas._libs.index.IndexEngine.get_loc
  File "index.pyx", line 196, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 7081, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 7089, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'synonym_case_mondo_is_many'
```

Reply:
Whaat! Very strange. I have been running this through the debugger, but also did a mini build, as well as a fresh build off an entirely new clone, and this did not happen.

Further strange, the stack trace doesn't show where in `sync_synonym.py` the error occurred.

Reply:
Just pushed a fix! Runs now.

The cause of this bug was a commit that I had just pushed to tentatively fix this "Bug: 3 instances where synonym column was mutated", but I was more focused on #724 which will both fix this bug as well as other issues. I hadn't bothered to fully check that this stand-in was running smoothly for all cases.

Reply:
Ok, I'll check again and report back.

Reply:
After the update, the make goal did run, but with some issues with https://github.com/monarch-initiative/mondo-ingest/pull/724 neither of these will be included for the January release.